### DISCUSSION TITLE
How to read variables in Lua filter?  pandoc 3.0

### DISCUSSION THREAD

**Discussion by mokeyish** [jgm/pandoc#9404] [Q&A]
> Accessing this variable in custom writers is deprecated. Starting with pandoc 3.0, it is set to a placeholder value (the default options) in custom writers. Access to the actual writer options is provided via the Writer or ByteStringWriter function, to which the options are passed as the second function argument.

I need to read a variable value set in the command line. Thank you for any assistance.

**Comment by lifeunleaded** [comment]
@mokeyish If you call pandoc with -M/--metadata, you can set a metadata value from the commandline which is accessible in the Lua filter.
E.g. 
`pandoc -M myvar=myval -f markdown -L my.lua -t native input.md`

With this in `input.md`
```
# My heading
My text
```

This in `my.lua`
```
function Meta (meta)
    myvar = meta.myvar
    return meta
end

function Header (hdr)
    hdr.identifier = myvar
    return hdr
end

return {
    { Meta = Meta },
    { Header = Header }
  }
```

Gives this result:
```
[ Header
    1
    ( "myval" , [] , [] )
    [ Str "My" , Space , Str "heading" ]
, Para [ Str "My" , Space , Str "text" ]
]
```

(note the command line variable passed as a metadata value using in the identifier, which was just the quickest example I could come up with, and note that you may need to ensure Meta is run before the function that uses it)
